client/js/store/map/Layers.js

```diff
@@ -271,13 +280,62 @@ const LayersStore = {
       }
     },
 
-    save ({ state, commit, dispatch }) {
-
-      return dpApi({
-        method: 'POST',
-        url: Routing.generate('api_resource_update', { resourceType: state.apiData.data.type, resourceId: state.apiData.data.id }),
-        data: { data: state.apiData }
+    saveAll ({ state, dispatch }) {
+      /* save each GIS layer and GIS layer category with its relationships */
+      state.apiData.included.forEach(el => {
+        dispatch('save', el)
       })
+    },
+
+    save ({ state, commit, dispatch }, resource) {
+      let payload
+
+      if (resource.type === 'GisLayer') {
+        payload = {
+          data: {
+            id: resource.id,
+            type: resource.type,
+            attributes: {
+              hasDefaultVisibility: resource.attributes.hasDefaultVisibility,
+              isMinimap: resource.attributes.isMinimap,
+              mapOrder: resource.attributes.mapOrder,
+              treeOrder: resource.attributes.treeOrder,
+              visibilityGroupId: resource.attributes.visibilityGroupId
+            },
+            relationships: {
+              parentCategory: {
+                data: {
+                  id: resource.attributes.categoryId,
+                  type: 'GisLayerCategory'
+                }
+              }
+            }
+          }
+        }
+      }
+
+      if (resource.type === 'GisLayerCategory') {
+        payload = {
+          data: {
+            id: resource.id,
+            type: resource.type,
+            attributes: {
+              treeOrder: resource.attributes.treeOrder,
+              hasDefaultVisibility: resource.attributes.hasDefaultVisibility,
+            },
+            relationships: {
+              parentCategory: {
+                data: {
+                  id: resource.attributes.parentId,
+                  type: 'GisLayerCategory'
+                }
+              }
+            }
+          }
+        }
+      }
```

Comment:
I prefer writing it like this, but it's just a suggestion

```suggestion
      let payload
      const { attributes, id, type } = resource
      const {
        categoryId
        hasDefaultVisibility,
        isMinimap,
        mapOrder,
        treeOrder,
        visibilityGroupId
      } = attributes

      if (resource.type === 'GisLayer') {
        payload = {
          data: {
            id,
            type,
            attributes: {
              hasDefaultVisibility,
              isMinimap,
              mapOrder,
              parentId,
              treeOrder,
              visibilityGroupId
            },
            relationships: {
              parentCategory: {
                data: {
                  id: categoryId,
                  type: 'GisLayerCategory'
                }
              }
            }
          }
        }
      }

      if (resource.type === 'GisLayerCategory') {
        payload = {
          data: {
            id,
            type,
            attributes: {
              treeOrder,
              hasDefaultVisibility,
            },
            relationships: {
              parentCategory: {
                data: {
                  id: parentId,
                  type: 'GisLayerCategory'
                }
              }
            }
          }
        }
      }
```

Reply:
you are right, done here https://github.com/demos-europe/demosplan-core/pull/3368/commits/452de7d66d36cc61be51e94a32dad3231f4aaa6b